### PR TITLE
Add WooCommerce flyout menu 

### LINKED ISSUE #6129: Fly-out menu items registration in WP menu
As a part of the new navigation updates, we're adding top-level core menu items back to the flyout menu as well as favorited extensions.

This issues will involve grabbing the top-level core items and registering those items in the WP admin menu under "WooCommerce" or if it's a category, linking to the category's first menu item.

screen-shot-2021-01-20-at-7 14 06-pm

## Deliverables 

* Marketplace is added to flyout
* Favorited extensions are added to the flyout

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #6129 

* Adds some helper methods to the menu
* Adds the top-level items from the main menu (with some exceptions from the design)

### Screenshots
Screen Shot 2021-01-22 at 6 09 17 PM

### Detailed test instructions:

1. Enable the new navigation.
2. Hover the "WooCommerce" menu item.
3. Check that expected items appear.
4. Check that clicking those items goes to the correct page.

### CODE REVIEW

File: src/Features/Navigation/CoreMenu.php [hunk 1]
```diff
@@ -276,6 +278,65 @@ public function register_post_types() {
...
+				$item['capability'],
+				isset( $item['url'] ) ? $item['url'] : null,
+				$item['title'],
+			);
+		}
+		// phpcs:enable
+
+		return $menu;
+	}
+
```
Comment (psealock) [inline src/Features/Navigation/CoreMenu.php:339]: Function looks good, the only thing is "Marketplace" is being added as well

**Comment by joshuatf** [comment]
Thanks for the heads up @psealock - updated to show the Marketplace item now.

**Review by psealock — APPROVED** [review]
This is looking great and working well. Pre-approving as there is one small nit-pick. The dark triangle at the top of the flyout needs to be revealed again. See Appearance:

Screen Shot 2021-02-02 at 2 20 35 PM


Screen Shot 2021-02-02 at 2 20 30 PM

**Comment by psealock** [comment]
🎉 Confirmed this is working with Nav Unification turned on

Screen Shot 2021-02-03 at 2 12 04 PM

**Comment by joshuatf** [comment]
Good catch, @psealock!  Removed the styles for non-wc pages in https://github.com/woocommerce/woocommerce-admin/pull/6165/commits/cdd5a130508b9f0bcd97d8a22d48b64c73023b08

**Pull request merged**